### PR TITLE
Update Azure PowerShell to use the new Common Authentication library

### PR DESCRIPTION
**Pull request by ogail**

### CODE REVIEW

File: src/Common/Commands.Common/AzurePowerShell.cs [hunk 1]
```diff
@@ -31,11 +31,11 @@ public class AzurePowerShell
 
         public const string AssemblyFileVersion = "0.9.7";
 
-        public const string ProfileFile = "AzureProfile.json";
+        public const string ProfileFile = "AzureSMProfile.json";
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePowerShell.cs]: I don't think we want to change this

File: src/Common/Commands.Common/AzurePowerShell.cs [hunk 2]
```diff
@@ -31,11 +31,11 @@ public class AzurePowerShell
...
 
-        public const string OldProfileFile = "WindowsAzureProfile.xml";
+        public const string OldProfileFile = "WindowsAzureSMProfile.xml";
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePowerShell.cs]: We definitely don't want to change this

File: src/Common/Commands.Common/AzurePowerShell.cs [hunk 3]
```diff
@@ -31,11 +31,11 @@ public class AzurePowerShell
...
 
-        public const string OldProfileFileBackup = "WindowsAzureProfile.xml.bak";
+        public const string OldProfileFileBackup = "WindowsAzureSMProfile.xml.bak";
```
Comment (markcowl) [inline src/Common/Commands.Common/AzurePowerShell.cs]: Or this

File: src/Common/Commands.Common/Commands.Common.csproj [hunk 4]
```diff
@@ -60,7 +60,7 @@
     </Reference>
     <Reference Include="Microsoft.Azure.Common.Authentication, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\packages\Microsoft.Azure.Common.Authentication.1.1.3-preview\lib\net45\Microsoft.Azure.Common.Authentication.dll</HintPath>
+      <HintPath>..\..\packages\Microsoft.Azure.Common.Authentication.1.1.4-preview\lib\net45\Microsoft.Azure.Common.Authentication.dll</HintPath>
```
Comment (markcowl) [inline src/Common/Commands.Common/Commands.Common.csproj]: 2.0

File: src/Common/Commands.Common/ProfileClient.cs [hunk 5]
```diff
@@ -0,0 +1,1158 @@
...
+                        .CreateCustomClient<SubscriptionClient>(
+                            new TokenCloudCredentials(commonTenantToken.AccessToken),
+                            environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ServiceManagement)))
+            {
+                var subscriptionListResult = SubscriptionClient.Subscriptions.List();
+                return subscriptionListResult.Subscriptions.Select(s => s.ActiveDirectoryTenantId).Distinct().ToArray();
+            }
+        }
+
+        private AzureSubscription MergeSubscriptionProperties(AzureSubscription subscription1, AzureSubscription subscription2)
```
Comment (markcowl) [inline src/Common/Commands.Common/ProfileClient.cs]: Remove

File: src/Common/Commands.Common/ProfileClient.cs [hunk 6]
```diff
@@ -0,0 +1,1158 @@
...
+                if (propertyValue != null)
+                {
+                    mergedEnvironment.Endpoints[property] = propertyValue;
+                }
+            }
+
+            return mergedEnvironment;
+        }
+
+        private AzureAccount MergeAccountProperties(AzureAccount account1, AzureAccount account2)
```
Comment (markcowl) [inline src/Common/Commands.Common/ProfileClient.cs]: Can't we remove this?

File: src/Common/Commands.Profile/Profile/ClearAzureProfile.cs [hunk 7]
```diff
@@ -25,13 +25,13 @@ namespace Microsoft.WindowsAzure.Commands.Profile
     /// <summary>
     /// Removes a Microsoft Azure environment.
     /// </summary>
-    [Cmdlet(VerbsCommon.Clear, "AzureProfile"), OutputType(typeof(AzureEnvironment))]
-    public class ClearAzureProfileCommand : SubscriptionCmdletBase
+    [Cmdlet(VerbsCommon.Clear, "AzureSMProfile"), OutputType(typeof(AzureEnvironment))]
```
Comment (markcowl) [inline src/Common/Commands.Profile/Profile/ClearAzureProfile.cs]: Should not change the cmdlet noun

File: src/Common/Commands.Profile/Profile/NewAzureProfile.cs [hunk 8]
```diff
@@ -32,8 +32,8 @@ namespace Microsoft.WindowsAzure.Commands.Profile
     /// <summary>
     /// Creates new Microsoft Azure profile.
     /// </summary>
-    [Cmdlet(VerbsCommon.New, "AzureProfile"), OutputType(typeof(AzureProfile))]
-    public class NewAzureProfileCommand : AzurePSCmdlet
+    [Cmdlet(VerbsCommon.New, "AzureSMProfile"), OutputType(typeof(AzureSMProfile))]
```
Comment (markcowl) [inline src/Common/Commands.Profile/Profile/NewAzureProfile.cs]: SHould not change the cmdlet noun

File: src/Common/Commands.Profile/Profile/SelectAzureProfile.cs [hunk 9]
```diff
@@ -32,14 +32,14 @@ namespace Microsoft.WindowsAzure.Commands.Profile
     /// <summary>
     /// Creates new Microsoft Azure profile.
     /// </summary>
-    [Cmdlet(VerbsCommon.Select, "AzureProfile"), OutputType(typeof(AzureProfile))]
-    public class SelectAzureProfileCommand : AzurePSCmdlet
+    [Cmdlet(VerbsCommon.Select, "AzureSMProfile"), OutputType(typeof(AzureSMProfile))]
```
Comment (markcowl) [inline src/Common/Commands.Profile/Profile/SelectAzureProfile.cs]: Cmdlet noun

**Comment by markcowl** [comment]
A few comments, mainly Looks good

File: src/Common/Commands.Common/AzurePowerShell.cs (hunk 1 above)
Comment (ogail) [inline src/Common/Commands.Common/AzurePowerShell.cs]: thanks for catching that

File: src/Common/Commands.Common/ProfileClient.cs (hunk 5 above)
Comment (ogail) [inline src/Common/Commands.Common/ProfileClient.cs]: We still need this as we merge between local subscription properties and remote subscription properties

File: src/Common/Commands.Common/ProfileClient.cs (hunk 6 above)
Comment (ogail) [inline src/Common/Commands.Common/ProfileClient.cs]: Same

**Comment by ogail** [comment]
retest this please

**Pull request merged**